plugins/out_td/td.c

```diff
@@ -269,6 +270,33 @@ int cb_td_exit(void *data, struct flb_config *config)
     return 0;
 }
 
+/* Configuration properties map */
+static struct flb_config_map config_map[] = {
+    {
+        FLB_CONFIG_MAP_STR, "api", NULL,
+        0, FLB_TRUE, 0, 
```

Comment:
the fifth parameter means "should I write the incoming value to the context?" , since this is FLB_TRUE and there is no offset in the context it will lead to a crash. Please fix the whole configmap.

make sure to write to the proper structure fields.